File: java/src/test/java/ProjectContext/pageObject/dealerPortal/dealerNewQuoteClientPage/Dealer_NewQuoteClientPage.java
package ProjectContext.pageObject.dealerPortal.dealerNewQuoteClientPage;


import core.components.buttons.buttonImpl;
import core.components.buttons.iButton;
import core.components.checkbox.checkboxImpl;
import core.components.checkbox.iCheckbox;
import core.components.dropdowns.dropdownImpl;
import core.components.dropdowns.iDropdown;
import core.components.input.iInput;
import core.components.input.inputImpl;
import org.openqa.selenium.By;

public class Dealer_NewQuoteClientPage {

    public Dealer_NewQuoteClientPage() {

    }

    public iInput txt_Email() {
        return new inputImpl(By.xpath("//*[@id='quote_client_email']"));
    }

    public iInput txt_First_name() {
        return new inputImpl(By.xpath("//*[@id='quote_client_firstName']"));
    }

    public iInput txt_Last_name() {
        return new inputImpl(By.xpath("//*[@id='quote_client_lastName']"));
    }

    public iInput txt_Secondary_owner_or_company() {
        return new inputImpl(By.xpath("//*[@id='quote_client_company']"));
    }

    public iInput txt_Distributor_name() {
        return new inputImpl(By.xpath("//*[@id='quote_client_distributorName']"));
    }

    public iInput txt_Address() {
        return new inputImpl(By.xpath("//*[@id='quote_client_address']"));
    }

    public iInput txt_Address2() {
        return new inputImpl(By.xpath("//*[@id='quote_client_address2']"));
    }

    public iInput txt_City() {
        return new inputImpl(By.xpath("//*[@id='quote_client_city']"));
    }

    public iDropdown dpd_Province() {
        return new dropdownImpl(By.xpath("//div[*[@for='quote_client_province']]/div"));
    }

    public iInput txt_Postal_code() {
        return new inputImpl(By.xpath("//*[@id='quote_client_postalCode']"));
    }

    public iInput txt_Tel_home() {
        return new inputImpl(By.xpath("//*[@id='quote_client_telHome']"));
    }

    public iInput txt_Tel_work() {
        return new inputImpl(By.xpath("//*[@id='quote_client_telWork']"));
    }

    public iInput txt_Indian_status_certificate() {
        return new inputImpl(By.xpath("//*[@id='quote_client_indianStatusCertificate']"));
    }

    public iInput txt_Drivers_license() {
        return new inputImpl(By.xpath("//*[@id='quote_client_driversLicense']"));
    }

    public iCheckbox chk_AggreePolicy() {
        return new checkboxImpl(By.xpath("//*[@id='quote_client_isAgreePolicy']"));
    }

    public iButton btn_Save() {
        return new buttonImpl(By.xpath("//*[@type='submit']"));
    }
}

File: playwright-automation/src/pages/dealerPortal/DealerNewQuoteClientPage.ts
import { Page, Locator, expect } from '@playwright/test';

/**
 * Quote Client data interface
 */
export interface QuoteClientData {
    email: string;
    firstName: string;
    lastName: string;
    secondaryOwnerOrCompany?: string;
    distributorName?: string;
    address?: string;
    address2?: string;
    city?: string;
    province?: string;
    postalCode?: string;
    telHome?: string;
    telWork?: string;
    indianStatusCertificate?: string;
    driversLicense?: string;
}

/**
 * Dealer New Quote Client Page - Create new quote with client info.
 * Converted from Dealer_NewQuoteClientPage.java + Dealer_NewQuoteClientPageFunc.java
 */
export class DealerNewQuoteClientPage {
    private page: Page;

    // Locators
    readonly txtEmail: Locator;
    readonly txtFirstName: Locator;
    readonly txtLastName: Locator;
    readonly txtSecondaryOwnerOrCompany: Locator;
    readonly txtDistributorName: Locator;
    readonly txtAddress: Locator;
    readonly txtAddress2: Locator;
    readonly txtCity: Locator;
    readonly dpdProvince: Locator;
    readonly txtPostalCode: Locator;
    readonly txtTelHome: Locator;
    readonly txtTelWork: Locator;
    readonly txtIndianStatusCertificate: Locator;
    readonly txtDriversLicense: Locator;
    readonly chkAgreePolicy: Locator;
    readonly btnSave: Locator;

    constructor(page: Page) {
        this.page = page;
        this.txtEmail = page.locator('#quote_client_email');
        this.txtFirstName = page.locator('#quote_client_firstName');
        this.txtLastName = page.locator('#quote_client_lastName');
        this.txtSecondaryOwnerOrCompany = page.locator('#quote_client_company');
        this.txtDistributorName = page.locator('#quote_client_distributorName');
        this.txtAddress = page.locator('#quote_client_address');
        this.txtAddress2 = page.locator('#quote_client_address2');
        this.txtCity = page.locator('#quote_client_city');
        this.dpdProvince = page.locator("//div[*[@for='quote_client_province']]/div");
        this.txtPostalCode = page.locator('#quote_client_postalCode');
        this.txtTelHome = page.locator('#quote_client_telHome');
        this.txtTelWork = page.locator('#quote_client_telWork');
        this.txtIndianStatusCertificate = page.locator('#quote_client_indianStatusCertificate');
        this.txtDriversLicense = page.locator('#quote_client_driversLicense');
        this.chkAgreePolicy = page.locator('#quote_client_isAgreePolicy');
        this.btnSave = page.locator('[type="submit"]');
    }

    /**
     * Fill new client form with provided data
     */
    async fillNewClientForm(client: QuoteClientData, submit: boolean = true): Promise<void> {
        await this.txtEmail.fill(client.email);
        await this.txtFirstName.fill(client.firstName);
        await this.txtLastName.fill(client.lastName);

        if (client.secondaryOwnerOrCompany) await this.txtSecondaryOwnerOrCompany.fill(client.secondaryOwnerOrCompany);
        if (client.distributorName) await this.txtDistributorName.fill(client.distributorName);
        if (client.address) await this.txtAddress.fill(client.address);
        if (client.address2) await this.txtAddress2.fill(client.address2);
        if (client.city) await this.txtCity.fill(client.city);
        if (client.province) {
            await this.dpdProvince.click();
            await this.page.locator(`li:has-text("${client.province}")`).click();
        }
        if (client.postalCode) await this.txtPostalCode.fill(client.postalCode);
        if (client.telHome) await this.txtTelHome.fill(client.telHome);
        if (client.telWork) await this.txtTelWork.fill(client.telWork);
        if (client.indianStatusCertificate) await this.txtIndianStatusCertificate.fill(client.indianStatusCertificate);
        if (client.driversLicense) await this.txtDriversLicense.fill(client.driversLicense);

        // Always check agree policy
        await this.chkAgreePolicy.check();

        if (submit) {
            await this.btnSave.click();
        }
    }

    /**
     * Click save button
     */
    async save(): Promise<void> {
        await this.btnSave.click();
    }
}
